src/Cli.Tests/UtilsTests.cs

```diff
@@ -200,6 +200,24 @@ public void TestApiPathIsWellFormed(string apiPath, ApiType apiType, bool expect
             );
         }
 
+        /// <summary>
+        /// Test to verify that when DAB_ENVIRONMENT variable is set, and dab-config.json and 
+        /// dab-config.{DAB_ENVIRONMENT}.json file is present, then when DAB engine is started, it will merge
+        /// the two config and use the merged config to startup the engine.
+        /// Here, dab-config.json has no connection_string, while dab-config.Test.json has a defined connection string.
+        /// once the `dab start` is executed the merge happens and the merged file contains the connection string from the
+        /// Test config. 
+        /// </summary>
+        [TestMethod]
+        public void TestMergeConfig()
+        {
+            Environment.SetEnvironmentVariable(RUNTIME_ENVIRONMENT_VAR_NAME, "Test");
+            File.WriteAllText("dab-config.json", CONFIG_TO_MERGE);
+            File.WriteAllText("dab-config.Test.json", CONFIG_TO_MERGE_WITH);
+            string mergedConfig = MergeConfigsIfAvailable();
+            Assert.IsTrue(JToken.DeepEquals(JObject.Parse(MERGED_CONFIG), JObject.Parse(File.ReadAllText(mergedConfig))));
```

Comment:
Youll need tests to cover a few other scenarios:
- merging of array's in addition to this test's confirmation of overriding the connection string. Can you clarify whether it is just the connection string that is overridden, or is technically the entire 
- non merging when a property in the override file is null, and the base file is non-null

Reply:
also when the override file has properties that the base file doesn't

Reply:
The test that is present already covers scenarios that shows it's the entire config merge. I'll update the comments so that's it's easier to note the difference.